src/adapters/lokijs/worker/executor.js

```diff
@@ -56,8 +56,36 @@ export default class LokiExecutor {
     await this._migrateIfNeeded()
   }
 
+  isCached(table: TableName<any>, id: RecordId): boolean {
+    const cachedSet = this.cachedRecords.get(table)
+    if (!cachedSet) {
+      return false
+    }
+
+    return cachedSet.has(id)
+  }
+
+  markAsCached(table: TableName<any>, id: RecordId): void {
+    let cachedSet = this.cachedRecords.get(table)
+    if (!cachedSet) {
+      cachedSet = new Set()
+      this.cachedRecords.set(table, cachedSet)
+    }
+
+    cachedSet.add(id)
```

Comment:
let's do it without local mutability:

cachedSet = xxxx

if ( cachedSet) {
cachedSet.add(x)
} else {
cachedRecords.set(table, new Set(x))